### PR TITLE
Laboratory work #2, Ivanov Artur - 20FPL1

### PR DESCRIPTION
**Pull request by ArturIvanov228**

### CODE REVIEW

File: lab_2/main.py [hunk 1]
```diff
@@ -33,7 +49,16 @@ def get_language_features(language_profiles: dict) -> list or None:
         and sorts them in alphabetical order
     :param language_profiles: a dictionary of dictionaries - language profiles
     """
-    pass
+    if not isinstance(language_profiles, dict):
+        return None
+    words = set()
```
Comment (ValeriyaKuznetsova) [inline lab_2/main.py:54]: what is a set?

File: lab_2/main.py [hunk 2]
```diff
@@ -33,7 +49,16 @@ def get_language_features(language_profiles: dict) -> list or None:
...
+    for item in language_profiles.items():
+        for word in item[1]:
```
Comment (ValeriyaKuznetsova) [inline lab_2/main.py:56]: why do we need items? and item[1]?

File: lab_2/main.py [hunk 3]
```diff
@@ -33,7 +49,16 @@ def get_language_features(language_profiles: dict) -> list or None:
...
+            words.add(word)
+    if not words:
+        return None
+    return sorted(list(words))
+
 
 
 def get_text_vector(original_text: list, language_profiles: dict) -> list or None:
```
Comment (ValeriyaKuznetsova) [inline lab_2/main.py:64]: one more function to implement

**Review by ValeriyaKuznetsova — COMMENTED** [review]
keep working! don't forget about start.py

File: lab_2/main.py [hunk 4]
```diff
@@ -7,6 +7,14 @@
 
 
 # 4
+def get_freq_dict(tokens: list) -> dict or None:
```
Comment (ValeriyaKuznetsova) [inline lab_2/main.py]: why add an empty function with the same name?

File: lab_2/main.py [hunk 5]
```diff
@@ -49,103 +58,37 @@ def get_language_features(language_profiles: dict) -> list or None:
...
+        if not isinstance(elem, str):
+            return None
+    features = get_language_features(language_profiles)
+    # original_text = set(original_text)
+    text_vector = []
+    for feature in features:
+        if feature not in original_text:
+            text_vector.append(0)
+        else:
+            text_vector.append(original_text.count(feature) / len(original_text))
```
Comment (ValeriyaKuznetsova) [inline lab_2/main.py:92]: looks like you did this here `get_freq_dict`

File: lab_2/start.py [hunk 6]
```diff
@@ -38,6 +44,24 @@
         UNKNOWN_SAMPLES = file_to_read.read().split('[TEXT]')[1:]
 
     EXPECTED = ['de', 'eng', 'lat']
-    RESULT = ''
+    RESULT = []
```
Comment (ValeriyaKuznetsova) [inline lab_2/start.py:47]: describe in writing what happens in `start.py`, please

File: lab_2/main.py (hunk 5 above)
Comment (ArturIvanov228) [reply lab_2/main.py:92]: get freq dict returns dict but we need a list

**Pull request closed without merging**